hazelcast/src/main/java/com/hazelcast/query/impl/Indexes.java

```diff
@@ -108,46 +109,32 @@ public static Builder newBuilder(SerializationService ss, IndexCopyBehavior inde
         return new Builder(ss, indexCopyBehavior);
     }
 
-    /**
-     * Obtains the existing index or creates a new one (if an index doesn't exist
-     * yet) for the given name in this indexes instance.
-     *
-     * @param name    the name of the index; the passed value might not
-     *                represent a canonical index name (as specified by
-     *                {@link Index#getName()}, in this case the method
-     *                canonicalizes it.
-     * @param ordered {@code true} if the new index should be ordered, {@code
-     *                false} otherwise.
-     * @param partitionStoreAdapter the reference to the store adapter. {@code null} if the index is global.
-     * @return the existing or created index.
-     */
-    public synchronized InternalIndex addOrGetIndex(String name, boolean ordered, StoreAdapter partitionStoreAdapter) {
-        InternalIndex index = indexesByName.get(name);
-        if (index != null) {
-            return index;
-        }
+    public synchronized InternalIndex addOrGetIndex(IndexConfig indexConfig, StoreAdapter partitionStoreAdapter) {
+        String name = indexConfig.getName();
 
-        String[] components = PredicateUtils.parseOutCompositeIndexComponents(name);
-        if (components == null) {
-            name = PredicateUtils.canonicalizeAttribute(name);
-        } else {
-            name = PredicateUtils.constructCanonicalCompositeIndexName(components);
-        }
+        assert name != null;
+        assert !name.isEmpty();
 
-        index = indexesByName.get(name);
+        InternalIndex index = indexesByName.get(name);
         if (index != null) {
             return index;
         }
 
-        index = indexProvider.createIndex(name, components, ordered, extractors, serializationService, indexCopyBehavior,
-                stats.createPerIndexStats(ordered, usesCachedQueryableEntries), partitionStoreAdapter);
+        index = indexProvider.createIndex(
+            indexConfig,
+            extractors,
+            serializationService,
+            indexCopyBehavior,
+            stats.createPerIndexStats(indexConfig.getType() == IndexType.SORTED, usesCachedQueryableEntries),
+            partitionStoreAdapter
+        );
 
         indexesByName.put(name, index);
         attributeIndexRegistry.register(index);
         converterCache.invalidate(index);
 
         indexes = indexesByName.values().toArray(EMPTY_INDEXES);
-        if (components != null) {
+        if (index.getComponents().length > 0) {
```

Comment:
Looks like it should be 1, not 0.

Reply:
Fixed in https://github.com/hazelcast/hazelcast/pull/15537/commits/2163cbdd5ac5290a80317aa92679e8ded7520b42
Thanks for spotting this.